components/request/request-controller.js

```diff
@@ -175,31 +242,59 @@ function factory($scope, $http, $location, ipCookie, config, brAlertService) {
       });
   };
 
-  function _navigateToIdp(session) {
+  self.showAll = function(identity, loginForm) {
+    identity.selected = false;
+    self.identitySelected = false;
+    loginForm.password.$setUntouched();
+    loginForm.$setPristine();
+  };
+
+  function _navigateToIdp(options) {
+    var session = options.session;
     var id = uuid.v4();
     var authioCallback =
       config.data.baseUri + '/credentials?id=' + id;
     var credentialCallback = $location.search().credentialCallback;
     var storageCallback = $location.search().storageCallback;
+    var credentialRequest = {};
+    var publicKeyType = ['CryptographicKey'];
+
+    if(self.keyType === keyTypes.TEMPORARY) {
+      publicKeyType.push('EmphemeralCryptographicKey');
+    }
+    if(self.keyType === keyTypes.NEW) {
+      publicKeyType.push('NewCryptographicKey');
```

Comment:
Don't add this type. Just later check for presence of `id`... no `id` means key is new or ephemeral, type will indicate whether it's ephemeral or not.